backend/typescript/services/implementations/EntityService.ts

```diff
@@ -0,0 +1,40 @@
+import { userInfo } from "node:os";
+import { Entity, IEntity } from "../../models/entity.mgmodel";
+import { IEntityService, EntityRequestDTO } from "../interfaces/IEntityService";
+var mongoose = require('mongoose');
+
+
+export class EntityService implements IEntityService {
+
+    /* retrieve the Entity with the given id */
+    async getEntity(id: string): Promise<IEntity> {
+        let entity: IEntity | null;
+        entity = await Entity.findById(mongoose.Types.ObjectId(id));
+
+        if (!entity) {
+            throw new Error;
```

Comment:
Give an error message, `id ${id} not found`. Also rebase on `main` after #6 gets merged and integrate the logger for error handling